esp-hal-common/src/serial.rs

```diff
@@ -167,20 +442,102 @@ pub trait Instance {
 
         idle
     }
+
+    fn configure_pins<
+        TX: OutputPin<OutputSignal = OutputSignal>,
+        RX: InputPin<InputSignal = InputSignal>,
+        CTS: InputPin<InputSignal = InputSignal>,
+        RTS: OutputPin<OutputSignal = OutputSignal>,
+    >(
+        &self,
+        pins: Option<Pins<TX, RX, CTS, RTS>>,
+    ) {
+        if pins.is_some() {
+            let pins = pins.unwrap();
+
+            if pins.tx.is_some() {
+                pins.tx
+                    .unwrap()
+                    .set_to_push_pull_output()
+                    .connect_peripheral_to_output(self.tx_signal());
+            }
```

Comment:
Using `if let Some(tx) = pins.tx {}` would eliminate the `.unwrap()` here. Not a huge deal but probably a bit more idiomatic.